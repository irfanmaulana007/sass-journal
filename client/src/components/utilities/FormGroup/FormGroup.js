import React from 'react'
import PropTypes from "prop-types"

const FormGroup = ({ type, label, name, handleChange, value }) => {
    const inputProps = {
        name: name,
        value: value,
        onChange: handleChange,
        autocomplete: 'off'
    }
    let inputType
    
    switch(type) {
    case 'text': inputType = <input className="form-control" {...inputProps} />
        break;
    case 'textarea': inputType = <textarea rows="5" className="form-control" {...inputProps}></textarea>
        break;
    default: inputType = <input className="form-control" type="text" {...inputProps} />
        break;
    }

    return (
        <div className="form-group">
            <label className="small text-uppercase text-muted">{label ? label: name}</label>
            {inputType}
        </div>
    )
}

FormGroup.propTypes = {
    label: PropTypes.string,
    type: PropTypes.string,
    name: PropTypes.string.isRequired,
    value: PropTypes.string,
    handleChange: PropTypes.func
}

export default FormGroup