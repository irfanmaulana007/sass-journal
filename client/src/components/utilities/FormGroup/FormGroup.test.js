import React from 'react'
import { shallow } from 'enzyme'

import FormGroup from './FormGroup'

const props = {
    label: 'Input Email',
    name: 'email'
}
const wrapped = shallow(<FormGroup label={props.label} name={props.name} />)

describe('FormGroup', () => {
    it('should render the FormGroup Component correctly', () => {
        expect(wrapped).toMatchSnapshot();
    });
    it('renders the FormGroup Label', () => { 
        expect(wrapped.find('label').text()).toEqual(props.label);
    });
})