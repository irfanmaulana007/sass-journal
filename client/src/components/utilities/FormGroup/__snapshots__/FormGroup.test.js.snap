// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup should render the FormGroup Component correctly 1`] = `ShallowWrapper {}`;
